### ISSUE TITLE
Expose DifferentiableGraphBackward to python

### ISSUE THREAD

**Issue by zasdfgbnm** [pytorch/pytorch#28549]
## 🚀 Feature

Currently, there is no easy way of visualizing the graph generated by autodiff in jit. We want something that could do

```python
import torch

@torch.jit.script
def f(a):
    return (a + 100) * 5

a = torch.tensor(0., requires_grad=True)
result = f(a)

print(result.grad_fn.graph)
```

## Additional context

In autograd, the `Node` for `grad_fn` of a JIT graph is defined as

```
struct DifferentiableGraphBackward : public autograd::Node
``` 
in `torch/csrc/jit/graph_executor.cpp`

This class does not have python correspondence, therefore python only views it as a general `CppFunction`. We need to create a Python class for it so that we could access additional information.

**Please confirm if this feature sounds useful and if yes, assign me to this issue and I will work on it.**

**Comment by ailzhang** [comment]
Note that the graph exposed here is not eager autograd but symbolic autodiff in JIT. But yea I think it's good to have if not much effort is necessary.

**Comment by zasdfgbnm** [comment]
@ailzhang Yes, I understand it is not eager autograd. I am looking at the performance on the backward pass with jit and feeling that this feature would be very helpful for debugging.

**Comment by albanD** [comment]
FYI: you can find [here](https://github.com/pytorch/pytorch/blob/master/torch/csrc/autograd/functions/init.cpp) an example of how to create a custom Node with additional available field.

**Comment by zasdfgbnm** [comment]
@albanD Thanks!

**Comment by ailzhang** [comment]
cc: @zdevito as well to see whether this fits the plan.

**Comment by zdevito** [comment]
Improving the debuggability of the JIT is a good thing. There are some things to keep in mind though when designing any interface around this. First, DifferentiableGraphBackward is an implementation detail of the JIT and should remain that way. It is used when part of the graph is symbolically differentiable, but there is no guarantee that a single call to the backward of the output of a script function will result in the execution of one DifferentiableGraphBackward op. In cases where none of the graph is symbolically differentiable, then there will be no instances of this op. In cases where the whole graph is symbolically differentiable, there will be a single one, and in partial cases, there will often be multiple instances of this op type in the graph.  In many cases, the final value of a script function won't have a `grad_fn` that is a DifferentiableGraphOp, though there may be autograd graph nodes inside the trace of that thing which do have these values. What we can do is improve how the `grad_fn` prints for when a DifferentiableGraphOp does exists. In this case, it would great if calling `print(x.grad_fn)` prints out the underlying graph as long as the connectivity information (the things inside the attributes of the Node* representing the op that show how the graph hooks up with captured variables). That would help understanding how the JIT constructed an autograd graph and show how it integrates into the existing autograd tape.

**Comment by janeyx99** [comment]
TorchScript has been deprecated in favor of torch.export as of 2.11. https://pytorch.org/blog/pytorch-2-11-release-blog/ Closing this issue as we are not further developing on it--please ping if you believe this issue was erroneously closed.
